core/src/main/java/smile/classification/IsotonicRegressionScaling.java

```diff
@@ -0,0 +1,220 @@
+/*******************************************************************************
+ * Copyright (c) 2010 Haifeng Li
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package smile.classification;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+import smile.sort.QuickSort;
+
+/**
+ * An alternative approach to probability calibration is to fit an isotonic
+ * regression model to an ill-calibrated probability model. This has been shown
+ * to work better than Platt scaling, in particular when enough training data is
+ * available.
+ * 
+ * <h2>References</h2>
+ * <ol>
+ * <li>Niculescu-Mizil, Alexandru; Caruana, Rich: Predicting Good Probabilities
+ * With Supervised Learning, 2005.</li>
+ * </ol>
+ *
+ * @author rayeaster
+ */
+public class IsotonicRegressionScaling implements Serializable {
+    private static final long serialVersionUID = 1L;
+    
+    private double trainMixScore = Double.NaN;
+    private double trainMaxScore = Double.NaN;
+
+    /** setep-wise constant functions */
+    private static class StepwiseConsFunc {
+        double lowScore;
+        double highScore;
+        double val;
+        int weight = 1;
+
+        StepwiseConsFunc next;
+        StepwiseConsFunc previous;
+
+        StepwiseConsFunc(double lowScore, double highScore, double val) {
+            this.lowScore = lowScore;
+            this.highScore = highScore;
+            this.val = val;
+        }
+
+        void setNext(StepwiseConsFunc next) {
```

Comment:
It is not necessary to use getter and setter for inner class. They make codes cluttered and don't really bring in benefits of encapsulations. You are the only user of this class!